keras_hub/src/models/retinanet/retinanet_label_encoder.py

```diff
@@ -0,0 +1,262 @@
+        """Creates box and classification targets for a batch.
+
+        Args:
+            images: a batched [batch_size, H, W, C] image float Tensor.
+            box_labels: a batched Keras style bounding box dictionary containing
+                bounding boxes and class labels. Should be in
+                `bounding_box_format`.
+
+        Returns:
+            encoded_box_targets: A tensor of shape (batch_size, num_anchors, 4)
+                containing the encoded box targets.
+            class_targets: A tensor of shape (batch_size, num_anchors, 1)
+                containing the class targets for each anchor.
+        """
+
+        if isinstance(images, tf.RaggedTensor):
+            raise ValueError(
+                "`RetinaNetLabelEncoder`'s `call()` method does not "
+                "support RaggedTensor inputs for the `images` argument. "
+                f"Received `type(images)={type(images)}`."
+            )
+
+        images_shape = ops.shape(images)
+        if len(images_shape) != 4:
+            raise ValueError(
+                "`RetinaNetLabelEncoder`'s `call()` method does not "
+                "support  unbatched inputs for the `images` argument. "
+                f"Received `shape(images)={images_shape}`."
+            )
+        image_shape = images_shape[1:]
+        bounding_boxes = to_dense(
+            bounding_boxes, max_boxes=self.max_dense_boxes
+        )
+
+        if len(ops.shape(bounding_boxes["classes"])) == 2:
+            bounding_boxes["classes"] = ops.expand_dims(
+                bounding_boxes["classes"], axis=-1
+            )
+
+        anchor_boxes = self.anchor_generator(images=images)
+        anchor_boxes = ops.concatenate(list(anchor_boxes.values()), axis=0)
+
+        result = self._encode_sample(bounding_boxes, anchor_boxes, image_shape)
+        encoded_box_targets = result["boxes"]
+        encoded_box_targets = ops.reshape(
+            encoded_box_targets, (-1, ops.shape(encoded_box_targets)[1], 4)
+        )
+        class_targets = result["classes"]
+        return encoded_box_targets, class_targets
+
+    def _encode_sample(self, bounding_boxes, anchor_boxes, image_shape):
+        """Creates box and classification targets for a batched sample.
+
+        Matches ground truth boxes to anchor boxes based on IOU.
+        1. Calculates the pairwise IOU for the M `anchor_boxes` and N `gt_boxes`
+            to get a `(M, N)` shaped matrix.
+        2. The ground truth box with the maximum IOU in each row is assigned to
+            the anchor box provided the IOU is greater than `match_iou`.
+        3. If the maximum IOU in a row is less than `ignore_iou`, the anchor
+            box is assigned with the background class.
+        4. The remaining anchor boxes that do not have any class assigned are
+            ignored during training.
+
+        Args:
+            gt_boxes: A float tensor with shape `(num_objects, 4)` representing
```

Comment:
doc string is not matching the args list

Reply:
Done